harness/determined/common/experimental/experiment.py

```diff
@@ -514,10 +516,12 @@ def get_with_offset(offset: int) -> bindings.v1GetExperimentCheckpointsResponse:
                 states=[bindings.checkpointv1State.COMPLETED],
             )
 
-        resps = api.read_paginated(
-            get_with_offset=get_with_offset,
-            pages=api.PageOpts.single if max_results else api.PageOpts.all,
-        )
+        with self._session as sess:
+            resps = api.read_paginated(
+                get_with_offset=get_with_offset,
+                pages=api.PageOpts.single if max_results else api.PageOpts.all,
+                session=sess,
```

Comment:
As currently written with separate `read_paginated` and `read_paginated_with_session`, I don't understand how this works. `session` isn't an argument of `read_paginated`.